tests/test_models.py

```diff
@@ -134,7 +139,105 @@ def test_build(simai_client):
     assert isinstance(launched_model.configuration, ModelConfiguration)
     assert launched_model.project_id == MODEL_RAW["project_id"]
 
-    model_conf = asdict(launched_model.configuration)
-    model_conf.pop("project_id")
+    assert launched_model.configuration._to_payload() == MODEL_CONF_RAW
 
-    assert model_conf == MODEL_CONF_RAW
+
+def test_set_doa(simai_client):
+    """WHEN the Domain of Analysis is updated
+    THEN the simulation_volume is updated accordingly.
+    """
+
+    model_conf = ModelConfiguration(project_id=MODEL_RAW["project_id"], **MODEL_CONF_RAW)
+
+    new_height = {"position": "relative_to_center", "value": 0.5, "length": 15.2}
+
+    model_conf.domain_of_analysis.height = DomainAxisDefinition(**new_height)
+
+    assert model_conf._to_payload().get("simulation_volume").get("Z").get("type") == new_height.get(
+        "position"
+    )
```

Comment:
nit but on top of being shorter, this also gives more meaningful errors (`KeyError: 'Z'` instead of `AttributeError: 'NoneType' object has no attribute 'get'`)
```suggestion
    assert model_conf._to_payload()["simulation_volume"]["Z"]["type"] == new_height["position"]
```